cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -196,20 +203,32 @@ struct MeanImpl : public SumImpl<ArrowType, SimdLevel> {
     }
     return Status::OK();
   }
+
+  Status Finalize(KernelContext*, Datum* out) override { return FinalizeImpl(out); }
+};
+
+template <typename ArrowType, SimdLevel::type SimdLevel>
+struct MeanImpl<ArrowType, SimdLevel,
+                std::enable_if_t<!is_decimal_type<ArrowType>::value>>
+    : public SumImpl<ArrowType, SimdLevel, DoubleType> {
```

Comment:
Can you add a comment above explaining that you're overriding the sum result type and why?